module/zfs/dbuf.c

```diff
@@ -1210,6 +1225,7 @@ dbuf_read_done(zio_t *zio, const zbookmark_phys_t *zb, const blkptr_t *bp,
 		ASSERT(db->db_blkid != DMU_BONUS_BLKID);
 		ASSERT3P(db->db_buf, ==, NULL);
 		db->db_state = DB_UNCACHED;
+		DTRACE_SET_STATE(db, "i/o error");
```

Comment:
I probably missed a review or discussion somewhere, but the original version of this change sets the dbuf state at the same time as generating the trace events.  Why did that change?

Reply:
If you look at previous comments you'll see that that was voted down By Brad and Brian.

Reply:
Thanks for the info.